cli.py

```diff
@@ -0,0 +1,562 @@
+                if not (set(keys) - set(data_keys)):
+                    sli_exists = True
+                    break
+
+    if not sli_exists:
+        raise SLRClientError(
+            'Some SLI keys do not exist. Please check the data returned from the check and the corresponding keys '
+            'in the data-source. Found the following keys returned from check {}: {}'.format(
+                check_id, set(sample_data)))
+
+
+def get_client(config):
+    return Client(config['url'], zign.api.get_token('uid', ['uid']))
+
+
+class SLRClientError(Exception):
+    pass
+
+
+class Client:
+    PRODUCT_GROUPS = 'product-groups'
+    PRODUCTS = 'products'
+    SLO = 'service-level-objectives'
+    DATA_SOURCE = 'data-sources'
+    SLI = 'service-level-indicators'
+
+    def __init__(self, url, token):
+        self.url = url
+        self.token = token
+
+        self.session = requests.Session()
+        self.session.headers.update({'Authorization': 'Bearer {}'.format(token), 'User-Agent': 'SLR-SLI/0.1-alpha'})
+
+    def endpoint(self, *args, trailing_slash=False, base_url=None):
+        parts = list(args)
+
+        # Ensure trailing slash!
+        if trailing_slash:
+            parts.append('')
+
+        url = self.url if not base_url else base_url
+
+        return urljoin(url, '/'.join(str(p).strip('/') for p in parts))
+
+    def product_list(self):
+        res = self.session.get(self.endpoint(self.PRODUCTS))
+        res.raise_for_status()
+
+        return res.json()
+
+    def product_delete(self, name):
+        res = self.session.delete(self.endpoint(self.PRODUCTS, name))
+        res.raise_for_status()
+
+        return res
+
+    def product_create(self, name, product_group):
+        data = {'name': name, 'product_group': product_group}
+
+        pgs = self.product_group_list()
+        match = [pg for pg in pgs if pg['slug'] == product_group]
+        if not match:
+            raise SLRClientError('Product group "{}" does not exist!'.format(product_group))
+
+        res = self.session.post(self.endpoint(self.PRODUCTS), json=data)
+        res.raise_for_status()
+
+        return res
+
+    def product_group_list(self):
+        res = self.session.get(self.endpoint(self.PRODUCT_GROUPS))
+        res.raise_for_status()
+
+        return res.json()
+
+    def product_group_delete(self, name):
+        res = self.session.delete(self.endpoint(self.PRODUCT_GROUPS, name))
+        res.raise_for_status()
+
+        return res
+
+    def product_group_create(self, name, department):
+        data = {
+            'name': name,
+            'department': department
+        }
+
+        res = self.session.post(self.endpoint(self.PRODUCT_GROUPS), json=data)
+        res.raise_for_status()
+
+        return res
+
+    def slo_list(self, product):
+        res = self.session.get(self.endpoint(self.SLO, product))
+        res.raise_for_status()
+
+        return res.json()
+
+    def slo_delete(self, product, id):
```

Comment:
SLOs are not identified by a name, only ID. The current API uses ID for deletion. The extra check is to avoid someone deleting and ID by mistake, this is why it requires the product although the product is not needed in the API call itself.